internal/compiler/object_tree.rs

```diff
@@ -1115,19 +1185,33 @@ impl Element {
                     },
                     &name_token,
                 );
+            } else if !lookup_result.is_local_to_component
+                && (lookup_result.property_visibility == PropertyVisibility::Private
+                    || lookup_result.property_visibility == PropertyVisibility::Output)
+            {
+                diag.push_error(
+                    format!(
+                        "assigning to {} property '{}'",
```

Comment:
Same here, I think the error message would benefit from a "reason" (even if it's just "can't" :)